constraint/pkg/externaldata/cache.go

```diff
@@ -101,15 +101,6 @@ func isValidCABundle(provider *unversioned.Provider) error {
 	}
 
 	switch u.Scheme {
-	case HTTPScheme:
-		if !provider.Spec.InsecureTLSSkipVerify {
-			return fmt.Errorf("only HTTPS scheme is supported for this Provider. To enable HTTP scheme, set insecureTLSSkipVerify to true")
-		}
-		if provider.Spec.CABundle != "" {
-			if err := verify(provider.Spec.CABundle); err != nil {
-				return err
-			}
-		}
 	case HTTPSScheme:
 		if provider.Spec.CABundle == "" {
 			return fmt.Errorf("caBundle should be set for HTTPS scheme")
```

Comment:
I do not understand why this is a requirement.  Normally when making a TLS connection with no pre-configured CA bundle, the system verifier is used (by leaving `tls.Config.RootCAs` as `nil`).

Reply:
let's handle this in a follow up. Opened #2452 for tracking